---
title: "Javi-plots-2"
output:
  pdf_document: default
  html_document: default
date: '2022-04-29'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries

```{r}
library("tidyverse")
```

# Loading Data

```{r}
#large_dataset <- read_tsv(file = "../data/02_large_w_meta_clean.tsv")
#treatment_dataset <- read_tsv(file = "../data/02_treatment_w_meta_clean.tsv")
#logfold_treatment <- read_tsv(file = "../data/03_treatment_log2fc.tsv")
combined_meta <- read_tsv(file = "../data/02_combined_meta.tsv")
combined_vst <- read_tsv(file = "../data/02_combined_vst.tsv.gz")
combined_dataset <- read_tsv(file = "../data/02_combined.tsv")
```

# Filter out combined data, keeping normal, pre, post and early RA
```{r}
normalized_dataset <- combined_vst %>% 
  filter(grepl("normal", id) | grepl("RA_", id)) %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "Reads") %>% 
  pivot_wider(names_from = id, values_from = Reads)

```


# Fold change normal with normal mean 
Using the mean of normal patients expression for each gene (Is this really 
an appropiate way to compare data?)

## Calculating the mean 
selecting only normal samples, transposing the dataset, calculating the mean 
for each gene

```{r}
normal_mean <- normalized_dataset %>% 
  select(Genes, starts_with("normal")) %>% 
  mutate(mean_reads = rowMeans(across(where(is.numeric)))) %>% 
  select(Genes, mean_reads)

normalized_dataset <- normalized_dataset %>% 
  left_join(normal_mean, by = "Genes")
```

## log fold change (using Mads code) 

**FC = sample/Normal_mean**

```{r}
only_logFCs <- normalized_dataset %>% 
  transmute(
    across(
      !c(Genes, mean_reads),
    ~ log2(. + 1) - log2(mean_reads + 1)
    )
  )

combined_logfc <- normalized_dataset %>% 
  select(Genes) %>% 
  bind_cols(only_logFCs) %>% 
  pivot_longer(cols = -c(Genes), names_to = "id", values_to = "logfc") %>% 
  pivot_wider(names_from = "Genes", values_from = "logfc") %>% 
  left_join(combined_meta, by = "id")

combined_logfc_ready <- combined_logfc %>% 
  filter(is.na(disease) | disease == "Normal" | disease == "Rheumatoid arthritis (established)") %>% 
  mutate(tag = case_when(
      treatment == TRUE ~ "RA post tDMARD",
      treatment == FALSE ~ "RA baseline",
      disease == "Normal" ~ "Normal",
      disease == "Rheumatoid arthritis (established)" ~ "RA established"
      )) %>% 
  relocate(tag, .after = id)

combined_logfc_ready

```

# Heatplot differential expression between Baseline RA and normal values

```{r}
combined_logfc_ready_long <- combined_logfc_ready %>% 
  select(-treatment, -disease_duration, -disease, -acc_num) %>% 
  pivot_longer(cols = -c(id, tag, sex, age),
               names_to = "Genes" , 
               values_to = "log2fc")

combined_logfc_ready_long %>% filter(log2fc >= 1 | log2fc <= -1) %>% 
  ggplot(mapping = aes(x = id,
                       y = Genes,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1)) 
```

# recreating Figure 2.B treatment paper

## Heatmap all genes starting with AA 
later decide how to filter: 
- which threshold is significant? 
- does the fold change of the post treatment expression has to be according to the normal mean or to the pre treatment for this plot?

```{r}
combined_logfc_ready_long %>% filter(tag == "RA established") %>% 
  filter(log2fc >= 1 | log2fc <= -1) %>% 
  ggplot(mapping = aes(x = id,
                       y = Genes,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  facet_wrap(~ tag, scales = 'free_x', ncol =) +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1))
```


# recreating Figure 2.B and 2.C treatment paper

## Just checking if the FC(baseline/normal) are upregulated as expected
```{r}
aux_dataset <- logfold_RA %>% 
  select(id, "CD3D", "CTLA4", "MS4A1", "CD19", "IL10", "MMP13", "CLEC12A", "CLEC2B", 
         "AURKA", "CD58") %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "log2fc")

aux_dataset %>%
  ggplot(mapping = aes(x = Genes,
                       y = id,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1)) 

```



# Boxplots with logfolds

```{r}
selected_logfold <-
  all_logfold %>% 
  filter( 
      Genes == "CD3D" 
    | Genes == "MS4A1"
    | Genes == "CTLA4"
    | Genes == "CD19"
    | Genes == "IL10"
    | Genes == "CLEC12A"
    | Genes == "AURKA"
    | Genes == "MMP13"
    | Genes == "CLEC2B"
    | Genes == "CD58")

selected_logfold %>% 
  ggplot(mapping = aes(
    x = Type,
    y = log2fc,
    colour = Type
  )
  ) +
  geom_boxplot() +
  geom_jitter()+
  facet_wrap(vars(Genes), scales = 'free_y', nrow = 2) +
  theme_bw() +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+
   
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1)) 
```

