---
title: "Javi-plots-2"
output:
  pdf_document: default
  html_document: default
date: '2022-04-29'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries

```{r}
library("tidyverse")
```

# Loading Data

```{r}
large_dataset <- read_tsv(file = "../data/02_large_w_meta_clean.tsv")
treatment_dataset <- read_tsv(file = "../data/02_treatment_w_meta_clean.tsv")
logfold_treatment <- read_tsv(file = "../data/03_treatment_log2fc.tsv")
```

# Fold change normal with normal mean 
Using the mean of normal patients expression for each gene (Is this really 
an appropiate way to compare data?)

## Calculating the mean 
selecting only normal samples, transposing the dataset, calculating the mean 
for each gene

```{r}
normal_dataset <- large_dataset %>% 
  filter(grepl("normal", id)) %>% 
  select(-disease, -acc_num, -sex, -age) %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "Reads") %>% 
  pivot_wider(names_from = id, values_from = Reads) %>% 
  mutate(mean_reads = rowMeans(across(where(is.numeric)))) 
```

## log fold change (using Mads code) 

**FC = Normal/Normal_mean**

```{r}
logfold_normal <- normal_dataset %>% 
  rowwise() %>% 
  mutate("Normal_2fc_1" = log2(normal_tissue_1+1)-log2(mean_reads+1),
         "Normal_2fc_2" = log2(normal_tissue_2+1)-log2(mean_reads+1),
         "Normal_2fc_3" = log2(normal_tissue_3+1)-log2(mean_reads+1),
         "Normal_2fc_4" = log2(normal_tissue_4+1)-log2(mean_reads+1),
         "Normal_2fc_5" = log2(normal_tissue_5+1)-log2(mean_reads+1),
         "Normal_2fc_6" = log2(normal_tissue_6+1)-log2(mean_reads+1),
         "Normal_2fc_7" = log2(normal_tissue_7+1)-log2(mean_reads+1),
         "Normal_2fc_8" = log2(normal_tissue_8+1)-log2(mean_reads+1),
         "Normal_2fc_9" = log2(normal_tissue_9+1)-log2(mean_reads+1),
         "Normal_2fc_10" = log2(normal_tissue_10+1)-log2(mean_reads+1),
         "Normal_2fc_11" = log2(normal_tissue_11+1)-log2(mean_reads+1),
         "Normal_2fc_12" = log2(normal_tissue_12+1)-log2(mean_reads+1),
         "Normal_2fc_13" = log2(normal_tissue_13+1)-log2(mean_reads+1),
         "Normal_2fc_14" = log2(normal_tissue_14+1)-log2(mean_reads+1),
         "Normal_2fc_15" = log2(normal_tissue_15+1)-log2(mean_reads+1),
         "Normal_2fc_16" = log2(normal_tissue_16+1)-log2(mean_reads+1),
         "Normal_2fc_17" = log2(normal_tissue_17+1)-log2(mean_reads+1),
         "Normal_2fc_18" = log2(normal_tissue_18+1)-log2(mean_reads+1),
         "Normal_2fc_19" = log2(normal_tissue_19+1)-log2(mean_reads+1),
         "Normal_2fc_20" = log2(normal_tissue_20+1)-log2(mean_reads+1),
         "Normal_2fc_21" = log2(normal_tissue_21+1)-log2(mean_reads+1),
         "Normal_2fc_22" = log2(normal_tissue_22+1)-log2(mean_reads+1),
         "Normal_2fc_23" = log2(normal_tissue_23+1)-log2(mean_reads+1),
         "Normal_2fc_24" = log2(normal_tissue_24+1)-log2(mean_reads+1),
         "Normal_2fc_25" = log2(normal_tissue_25+1)-log2(mean_reads+1),
         "Normal_2fc_26" = log2(normal_tissue_26+1)-log2(mean_reads+1),
         "Normal_2fc_27" = log2(normal_tissue_27+1)-log2(mean_reads+1),
         "Normal_2fc_28" = log2(normal_tissue_28+1)-log2(mean_reads+1)) %>% 
  select(-starts_with("normal_t"), -mean_reads) %>% 
  pivot_longer(cols = starts_with("Normal_2fc"), names_to = "id") %>% 
  pivot_wider(names_from = "Genes", values_from = "value")
```

# Fold change normal with baseline RA (RA_pre)

## Join RA baseline and normal data
```{r}
normal_mean <- normal_dataset %>% 
  select(Genes, mean_reads) %>% 
  pivot_longer(cols = -c(Genes),
               names_to = "id" , 
               values_to = "Reads") %>% 
  relocate(id, .before = Genes)

aux_dataset <- treatment_dataset %>% 
  select(-treatment, -sex, -age, -acc_num, -disease_duration) %>% 
  filter(grepl("RA_pre", id)) %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "Reads") %>% 
  bind_rows(normal_mean) %>% 
  pivot_wider(names_from = id, values_from = Reads)

```

## log fold change (using Mads code) 

**FC = Baseline_RA/Normal**

```{r}
logfold_RA <- aux_dataset %>% 
  rowwise() %>% 
  mutate("RAb_2fc_1" = log2(RA_pre_1+1)-log2(mean_reads+1),
         "RAb_2fc_2" = log2(RA_pre_2+1)-log2(mean_reads+1),
         "RAb_2fc_3" = log2(RA_pre_3+1)-log2(mean_reads+1),
         "RAb_2fc_4" = log2(RA_pre_4+1)-log2(mean_reads+1),
         "RAb_2fc_5" = log2(RA_pre_5+1)-log2(mean_reads+1),
         "RAb_2fc_6" = log2(RA_pre_6+1)-log2(mean_reads+1),
         "RAb_2fc_7" = log2(RA_pre_7+1)-log2(mean_reads+1),
         "RAb_2fc_8" = log2(RA_pre_8+1)-log2(mean_reads+1),
         "RAb_2fc_9" = log2(RA_pre_9+1)-log2(mean_reads+1),
         "RAb_2fc_10" = log2(RA_pre_10+1)-log2(mean_reads+1),
         "RAb_2fc_11" = log2(RA_pre_11+1)-log2(mean_reads+1),
         "RAb_2fc_12" = log2(RA_pre_12+1)-log2(mean_reads+1),
         "RAb_2fc_13" = log2(RA_pre_13+1)-log2(mean_reads+1),
         "RAb_2fc_14" = log2(RA_pre_14+1)-log2(mean_reads+1),
         "RAb_2fc_15" = log2(RA_pre_15+1)-log2(mean_reads+1),
         "RAb_2fc_16" = log2(RA_pre_16+1)-log2(mean_reads+1),
         "RAb_2fc_17" = log2(RA_pre_17+1)-log2(mean_reads+1),
         "RAb_2fc_18" = log2(RA_pre_18+1)-log2(mean_reads+1),
         "RAb_2fc_19" = log2(RA_pre_19+1)-log2(mean_reads+1)) %>% 
  select(-starts_with("RA_p"), -mean_reads) %>% 
  pivot_longer(cols = starts_with("RAb_2fc"), names_to = "id") %>% 
  pivot_wider(names_from = "Genes", values_from = "value")

```

# Heatplot differential expression between Baseline RA and normal values

```{r}
logfold_RA_long <- logfold_RA %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "log2fc")

logfold_RA_long %>% filter(log2fc >= 8 | log2fc <= -8) %>% 
  ggplot(mapping = aes(x = id,
                       y = Genes,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1)) 
```

# recreating Figure 2.B treatment paper

## Making tables longer
```{r}
logfold_normal_long <- logfold_normal %>% 
  mutate(Type = "Normal") %>% 
  relocate(Type, .after = id) %>% 
  pivot_longer(cols = -c(id, Type),
               names_to = "Genes" , 
               values_to = "log2fc")

logfold_RA_long <- logfold_RA%>% 
  mutate(Type = "RA baseline") %>% 
  relocate(Type, .after = id) %>% 
  pivot_longer(cols = -c(id, Type),
               names_to = "Genes" , 
               values_to = "log2fc")

logfold_treatment_long <- logfold_treatment%>% 
  mutate(Type = "RA post-tDMARD") %>% 
  relocate(Type, .after = id) %>% 
  pivot_longer(cols = -c(id, Type),
               names_to = "Genes" , 
               values_to = "log2fc")
```

## joining them 

```{r}
all_logfold <- logfold_normal_long %>% 
  bind_rows(logfold_RA_long) %>% 
  bind_rows(logfold_treatment_long)
```

## Heatmap all genes starting with AA 
later decide how to filter: 
- which threshold is significant? 
- does the fold change of the post treatment expression has to be according to the normal mean or to the pre treatment for this plot?

```{r}
all_logfold %>% filter(grepl("^AA", Genes)) %>% 
  ggplot(mapping = aes(x = id,
                       y = Genes,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  facet_wrap(~ Type, scales = 'free_x') +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1))
```


# recreating Figure 2.B and 2.C treatment paper

## Just checking if the FC(baseline/normal) are upregulated as expected
```{r}
aux_dataset <- logfold_RA %>% 
  select(id, "CD3D", "CTLA4", "MS4A1", "CD19", "IL10", "MMP13", "CLEC12A", "CLEC2B", 
         "AURKA", "CD58") %>% 
  pivot_longer(cols = -c(id),
               names_to = "Genes" , 
               values_to = "log2fc")

aux_dataset %>%
  ggplot(mapping = aes(x = Genes,
                       y = id,
                       fill = log2fc )) +
  geom_tile(alpha = 0.5) +
  scale_fill_gradient2(low = "blue",
                        mid = "white",
                        high = "red",
                        midpoint = 0) +
  theme_classic(base_size = 8) +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 45,
                                   hjust = 1)) 

```

